### PR TITLE
[api] Refactor option assignation as in PR review

### PR DESCRIPTION
**Pull request by Ana06**
The PR was merged before I submit the review: https://github.com/openSUSE/open-build-service/pull/5161#pullrequestreview-130364161 :sob:

### CODE REVIEW

File: src/api/app/models/package.rb [hunk 1]
```diff
@@ -929,8 +929,8 @@ def buildresult(prj = project, show_all = false)
   end
 
   def jobhistory_list(project, repository, arch, options = {})
-    options[:limit] = 100 if options[:limit].blank?
-    options[:package] = name if options[:package].blank?
+    options[:limit] |= 100
+    options[:package] |= name
```
Comment (ChrisBr) [inline src/api/app/models/package.rb]: This does not work when it is an empty string, does it?

Comment (Ana06) [reply src/api/app/models/package.rb]: is `package[:name]` is `""` it is imposible that you have the instance package from that name... :no_mouth:

Comment (ChrisBr) [reply src/api/app/models/package.rb]: This should be ``||=``

Comment (ChrisBr) [reply src/api/app/models/package.rb]: Maybe we could even drop the limit as it is nowhere used currently...

File: src/api/app/models/package.rb [hunk 2]
```diff
@@ -929,8 +929,8 @@ def buildresult(prj = project, show_all = false)
   end
 
   def jobhistory_list(project, repository, arch, options = {})
-    options[:limit] = 100 if options[:limit].blank?
-    options[:package] = name if options[:package].blank?
+    options[:limit] ||= 100
+    options[:package] ||= name
```
Comment (ChrisBr) [inline src/api/app/models/package.rb]: Why is this necessary @DavidKang ? The package name is set in the before filter? How can it be blank?

Comment (DavidKang) [reply src/api/app/models/package.rb]: True, it will ever exist

File: src/api/app/models/package.rb (hunk 1 above)
Comment (DavidKang) [reply src/api/app/models/package.rb]: IIRC, we added `limit` to be able to change it in the future. However, if the limit in the `backend` by default is 100 we could drop it

File: src/api/app/models/package.rb (hunk 2 above)
Comment (DavidKang) [reply src/api/app/models/package.rb]: @ChrisBr, we could remove it :+1:

Comment (Ana06) [reply src/api/app/models/package.rb]: Then I just remove both parameters, right? 😂😅

Comment (ChrisBr) [reply src/api/app/models/package.rb]: I think so, yes!

**Comment by Ana06** [comment]
@ChrisBr please re-review it  :wink:

**Comment by coolo** [comment]
If this is a class method I wonder why it's in the package model at all. Sounds like something that should be in the Backend namespace

File: src/api/app/models/package.rb [hunk 3]
```diff
@@ -929,12 +929,10 @@ def buildresult(prj = project, show_all = false)
   end
 
   def jobhistory_list(project, repository, arch, options = {})
-    options[:limit] = 100 if options[:limit].blank?
     options[:package] = name if options[:package].blank?
 
     results = Jobhistory.find_hashed(project: project.name, package: options[:package],
-                                     repository: repository, arch: arch,
-                                     limit: options[:limit])
```
Comment (bgeuken) [inline src/api/app/models/package.rb]: Did you verify that the default limit is 100?

File: src/api/app/models/package.rb [hunk 4]
```diff
@@ -929,8 +929,6 @@ def buildresult(prj = project, show_all = false)
   end
 
   def jobhistory_list(project, repository, arch, options = {})
-    options[:package] = name if options[:package].blank?
-
```
Comment (bgeuken) [inline src/api/app/models/package.rb]: So you could drop the whole options and make this a parameter or keyword parameter.

File: src/api/app/models/package.rb [hunk 5]
```diff
@@ -928,8 +928,8 @@ def buildresult(prj = project, show_all = false)
     LocalBuildResult::ForPackage.new(package: self, project: prj, show_all: show_all)
   end
 
-  def jobhistory_list(project, repository, arch, options = {})
-    results = Jobhistory.find_hashed(project: project.name, package: options[:package],
+  def jobhistory_list(project, repository, arch, package_name)
+    results = Jobhistory.find_hashed(project: project.name, package: package_name,
```
Comment (bgeuken) [inline src/api/app/models/package.rb]: Please squash this commit into the previous one.

File: src/api/spec/models/package_spec.rb [hunk 6]
```diff
@@ -401,7 +401,8 @@
   describe '#jobhistory_list' do
     let(:backend_url) { "#{CONFIG['source_url']}/build/#{home_project}/openSUSE_Tumbleweed/x86_64/_jobhistory?limit=100&package=#{package}" }
 
-    subject { package.jobhistory_list(home_project, 'openSUSE_Tumbleweed', 'x86_64') }
+    # the package name is needed because it could be a multibuild
+    subject { package.jobhistory_list(home_project, 'openSUSE_Tumbleweed', 'x86_64', package.name) }
```
Comment (bgeuken) [inline src/api/spec/models/package_spec.rb]: Looks like this broke with your previous change ("[api] package as param instead of an options hash"). Please squash them together

File: src/api/app/models/package.rb (hunk 3 above)
Comment (Ana06) [reply src/api/app/models/package.rb]: what do you mean?

File: src/api/app/models/package.rb (hunk 4 above)
Comment (Ana06) [reply src/api/app/models/package.rb]: that is what I did, didn't I?

File: src/api/app/models/package.rb (hunk 5 above)
Comment (Ana06) [reply src/api/app/models/package.rb]: why?

File: src/api/spec/models/package_spec.rb [hunk 7]
```diff
@@ -398,10 +398,11 @@
     end
   end
 
-  describe '#jobhistory_list' do
+  describe '::jobhistory_list' do
```
Comment (ChrisBr) [inline src/api/spec/models/package_spec.rb]: [Nitpick] everywhere else we use ``.`` instead of ``::`` for class methods.

**Comment by dmarcoux** [comment]
@Ana06 Still relevant?

**Comment by Ana06** [comment]
@dmarcoux yes, @bgeuken added some comments I am waiting for an answer... @coolo asked to move it somewhere else, but this is already an improvement

File: src/api/spec/models/package_spec.rb (hunk 7 above)
Comment (Ana06) [reply src/api/spec/models/package_spec.rb]: why do we use `.`? :confused:

Comment (bgeuken) [reply src/api/spec/models/package_spec.rb]: I would prefer `::` as well since that's what is usually used class methods. But one way or the other we should align this to one way of writing this.

File: src/api/spec/models/package_spec.rb (hunk 6 above)
Comment (bgeuken) [reply src/api/spec/models/package_spec.rb]: This is still not squashed:-)

File: src/api/app/models/package.rb (hunk 4 above)
Comment (bgeuken) [reply src/api/app/models/package.rb]: I was reviewing per commits and didn't see that at that time:wink:

File: src/api/app/models/package.rb (hunk 5 above)
Comment (bgeuken) [reply src/api/app/models/package.rb]: Because the previous commit is just an interim change that is obsoleted by this commit.

**Comment by bgeuken** [comment]
Oh, come on. If you really waited for an answer you should have pinged me:laughing::wink:

File: src/api/app/models/package.rb (hunk 3 above)
Comment (bgeuken) [reply src/api/app/models/package.rb]: Before your change the frontend would always set it to 100. Now that this is gone what default limit would the backend use?

Comment (Ana06) [reply src/api/app/models/package.rb]: now I get it, I am removing the limit :see_no_evil:

**Comment by Ana06** [comment]
@bgeuken 

> Oh, come on. If you really waited for an answer you should have pinged me

I didn't want to stress you :eyes:  I waited for an answer and at some point I forgot this PR... :unamused: Thanks for answering now! :wink:

**Comment by coolo** [comment]
I personally find 7 commits for 7 lines a little squashable

**Comment by bgeuken** [comment]
Okay, but there are still some tests to fix:-)

File: src/api/spec/models/package_spec.rb (hunk 7 above)
Comment (Ana06) [reply src/api/spec/models/package_spec.rb]: Here you go: https://github.com/openSUSE/open-build-service/pull/5775 :wink:

**Comment by Ana06** [comment]
@coolo @bgeuken please re-review it!

**Comment by Ana06** [comment]
Oh no! merge conflicts! :sob:

**Comment by Ana06** [comment]
@coolo 

> I guess the cassette updates are not necessary, but not important

they are. Remove them and Travis will get angry :angry:

**Comment by coolo** [comment]
But the good news is: circleci is not such a llorón - https://circleci.com/gh/coolo/open-build-service/tree/Ana06-options is all green without cassette updates.

And as we in Germany say in such occasions: *ÄTSCH BÄTSCH*

On a more serious note: One more mark in my 'Cassettes are better to be removed as about no OBS developer knows how to handle them correctly' rifle.

**Comment by Ana06** [comment]
@coolo 

> @Ana06 - you have to face the truth at some point: Travis is gone! Travis is pushing up the daisies, it run down the curtain and joined the bleedin' choir invisibile! Travis is an EX-CI!
> 
> And what makes things worse: you're still in phase 1 of 

Phase 2: :angry: :rage: :angry: 

@coolo 

> But the good news is: circleci is not such a llorón - https://circleci.com/gh/coolo/open-build-service/tree/Ana06-options is all green without cassette updates.

They were needed when I changed the name of the method to use `::` the cassettes names changed. But using `.` instead keep the same name. So you are right, they are not needed now, but they were needed when I first generated them... :unamused: 

> Cassettes are better to be removed as about no OBS developer knows how to handle them correctly

That's rude... Some of them know... :unamused:

**Comment by hennevogel** [comment]
>> Cassettes are better to be removed as about no OBS developer knows how to handle them correctly

> That's rude... Some of them know... unamused

All of them can know :smiling_imp:
